### ISSUE TITLE
@oas-tools/core typescript support

### ISSUE THREAD

**Issue by agnoam** [oas-tools/oas-tools#350]
At the time of this issue, `@oas-tools/core` does not support typescript at all. My suggestion is to add Typescript support.

I already have a draft for a `d.ts` file, it's still a work in progress. but it can be useful.

**Comment by alesancor1** [comment]
I completely agree with you, TS support is pending and is already planned for a future version. In fact, if you check the OAS Tools' Github project you'll find a draft on this topic.

I am not really familiar with ts, so any help I can have is always welcome :)

**Comment by iofirag** [comment]
any news about typescript support..?

**Comment by alesancor1** [comment]
Still planned for v3.1. I'm currently working in other projects and cannot dedicate as much time as I'd like to oas-tools, but it will come asap :-)

**Comment by agnoam** [comment]
Any news when should it be?

**Comment by alesancor1** [comment]
TS support will be delayed for a future v4.0.0. Some issues regarding backwards compatibility has been fixed, so `v3.1.0` will still be backwards compatible. After migrating the codebase to TS, some imports will be affected and it will be no longer compatible with v2.X, so I'm leaving this issue for the next major version.

**Comment by iofirag** [comment]
?

**Comment by iofirag** [comment]
anybody know about another package that worked in TS untill this one will fixed?

**Comment by shuat** [comment]
Anyone looking for a partial typescript definition you can add the following to a global.d.ts file and should work. If you have a src directory, than probably belongs in the root of that folder.

```
interface ConfigOptions {
  middleware: {
    validator: {
      requestValidation?: boolean;
      responseValidation?: boolean;
      strict?: boolean;
    };
    router: {
      disable?: boolean;
      controllers: string;
    };
    error: {
      disable?: boolean;
      printStackTrace?: boolean;
      customHandler?: ((...args: any[]) => void) | null; // Replace any[] with the actual argument types if possible
    };
    security: {
      disable?: boolean;
      auth?: any; // You may want to specify a more detailed type here
    };
    swagger: {
      disable?: boolean;
      path?: string;
      ui?: any; // You may want to specify a more detailed type here
    };
  };
  oasFile: string;
  logger: {
    level: boolean | string;
  };
}
declare module "@oas-tools/core" {
  function initialize(
    expressApp: import("express").Express,
    initialize: ConfigOptions,
  ): Promise<void>;
}
```

**Comment by shuat** [comment]
BTW I published a fork that solves this as well as default exports https://www.npmjs.com/package/@bitbeanllc/oas-tools
